### PR TITLE
Global Cors Configuration

### LINKED ISSUE #1566: Provide CORS Configurations Globally
### Describe your problem(s)
The user should have the capability to provide the CORS Configurations globally.

### Describe your solution
The config.toml should have a segment for the CORS configuration.

### How will you implement it

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
Provide CORS Global Configuration.
If an API Level Cors Configuration is not found, the global configuration will appear.

### Issues

Fixes #1566

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: Yes

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: integration/test-integration/src/test/resources/apim/config.toml [hunk 1]
```diff
@@ -35,6 +35,14 @@ clusterTimeoutInSeconds = 20
 certPath = "/home/wso2/security/keystore/mg.pem"
 keyPath = "/home/wso2/security/keystore/mg.key"
 
+[router.cors]
+enabled = false
+allowOrigins = ["*"]
+allowMethods = ["GET","PUT","POST","DELETE","PATCH","OPTIONS"]
+allowHeaders = ["authorization","Access-Control-Allow-Origin","Content-Type","SOAPAction","apikey", "testKey"]
```
Comment (HiranyaKavishani) [inline integration/test-integration/src/test/resources/apim/config.toml:42]: shall we add internal-key as well?

Comment (VirajSalaka) [reply integration/test-integration/src/test/resources/apim/config.toml]: added via https://github.com/wso2/product-microgateway/pull/1792/commits/3ac56472dcabc5b5e18837bf3442fa5fb021536b

File: adapter/internal/oasparser/model/mgw_swagger.go [hunk 2]
```diff
@@ -345,13 +347,31 @@ func (swagger *MgwSwagger) setXWso2Cors() {
...
 			return
 		}
 		logger.LoggerOasparser.Errorf("Error while parsing %v .", xWso2Cors)
+	} else {
+		swagger.xWso2Cors = generateGlobalCors()
 	}
+}
 
+func generateGlobalCors() *CorsConfig {
+	conf, _ := config.ReadConfigs()
```
Comment (dinusha92) [inline adapter/internal/oasparser/model/mgw_swagger.go:365]: shouldn't we handle the error here instead of ignoring it?

Comment (VirajSalaka) [reply adapter/internal/oasparser/model/mgw_swagger.go:365]: actually at this point, the config is already read (Coz this is a once operation). We should refactor the readConfigs methods I suppose.

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**